modules/collection/programs/fish.nix

```diff
@@ -0,0 +1,191 @@
+        end
+      ''
+      else strOrPath
+    ) "${funcName}.fish";
+
+  isVendored = plugin:
+    any (p: pathExists "${plugin}/share/fish/${p}") [
+      "vendor_conf.d"
+      "vendor_completions.d"
+      "vendor_functions.d"
+    ];
+in {
+  options.rum.programs.fish = {
+    enable = mkEnableOption "fish";
+
+    package = mkPackageOption pkgs "fish" {};
+
+    config = mkOption {
+      default = null;
+      type = nullOr (either str path);
+      description = ''
+        The main configuration for fish, written verbatim to `.config/fish/config.fish`.
+      '';
+    };
+
+    prompt = mkOption {
+      default = null;
+      type = nullOr (either str path);
+      description = ''
+        Your fish prompt, written to `.config/fish/functions/fish_prompt.fish`.
+        It follows the behaviour of `rum.programs.fish.functions`.
+      '';
+    };
+
+    functions = mkOption {
+      default = {};
+      type = attrsOf (oneOf [str path]);
+      description = ''
+        A fish function which is being written to `.config/fish/functions/<name>.fish`.
+
+        If the input value is a string, its contents will be wrapped
+        inside of a function declaration, like so:
+        ```fish
+            function <name>;
+                <function body>
+            end
+        ```
+
+        Otherwise you are expected to handle that yourself.
+      '';
+      example = literalExample ''
+        {
+          fish_prompt = pkgs.writers.writeFish "fish_prompt.fish" '\'
+              function fish_prompt -d "Write out the prompt"
+                    # This shows up as USER@HOST /home/user/ >, with the directory colored
+                    # $USER and $hostname are set by fish, so you can just use them
+                    # instead of using `whoami` and `hostname`
+                    printf '%s@%s %s%s%s > ' $USER $hostname \
+                        (set_color $fish_color_cwd) (prompt_pwd) (set_color normal)
+              end
+          '\';
+          hello-world = '\'
+              echo Hello, World!
+          '\';
+        }'';
+    };
+
+    earlyConfigFiles = mkOption {
+      default = {};
+      type = attrsOf (oneOf [str path]);
+      description = ''
+        Extra configuration files, they will all be written verbatim
+        to `.config/fish/conf.d/<name>.fish`.
+
+        Those files are run before `.config/fish/config.fish` as per the fish
+        [documentation](https://fishshell.com/docs/current/language.html#configuration-files).
+      '';
+      example = {
+        my-aliases = ''
+          alias la "ls -la"
+          alias ll "ls -l"
+        '';
+      };
+    };
+
+    abbrs = mkOption {
+      default = {};
+      type = attrsOf str;
+      description = ''
+        A set of fish abbreviations, they will be set up with the `abbr --add` fish builtin.
+      '';
+    };
+
+    plugins = mkOption {
+      type = attrsOf path;
```

Comment:
I would do a list of plugins instead. I assume `fetchFromGithub` creates a derivation with the name being the repo name, correct? We could use that instead, and this would simplify the parsing function too.

Reply:
Unfortunately the name of a gh derivation is "source"....

Reply:
Ah that's fine then.